### ISSUE TITLE
scipy.spatial.ConvexHull exit with code 134, free(): invalid pointer

### ISSUE THREAD

**Issue by ncarrara** [scipy/scipy#9852]
I try to compute a convex hulls on sets of points. For this particular set, python raise a free() : invalid pointer. Since there a many repeated row in the array, I've also tried with np.unique, and the same exception is raised. 

By the way, the exception is not always "free(): invalid pointer". If I add some breakline in the python file,I got Segmentation fault instead of "free() invalid pointer". 

### Reproducing code example:

```
import numpy as np
from scipy.spatial import ConvexHull

a = np.array([
    [5.83e-15,  9.34e-02],
    [2.32e-16, - 1.64e-18],
    [1.59e-30, - 7.92e-13],
    [-9.39e-23,  3.51e-19],
    [5.83e-15,  9.34e-02],
    [2.32e-16, - 1.64e-18],
    [1.59e-30, - 7.92e-13],
    [-9.39e-23,  3.51e-19],
    [5.83e-15,  9.34e-02],
    [2.32e-16, - 1.64e-18],
    [1.59e-30, - 7.92e-13],
    [-9.39e-23,  3.51e-19],
    [5.83e-15,  9.34e-02],
    [2.32e-16 ,- 1.64e-18],
    [1.59e-30, - 7.92e-13],
    [-9.39e-23,  3.51e-19],
    [5.83e-15,  9.34e-02],
    [2.32e-16 ,- 1.64e-18],
    [1.59e-30 ,- 7.92e-13],
    [-9.39e-23,  3.51e-19],
    [5.83e-15,  9.34e-02],
    [2.32e-16, - 1.64e-18],
    [1.59e-30, - 7.92e-13],
    [-9.39e-23,  3.51e-19],
    [5.83e-15,  9.34e-02],
    [2.32e-16 ,- 1.64e-18],
    [1.59e-30 ,- 7.92e-13],
    [-9.39e-23,  3.51e-19],
    [5.83e-15 , 9.34e-02],
    [2.32e-16, - 1.64e-18],
    [1.59e-30 ,- 7.92e-13],
    [-9.39e-23,  3.51e-19],
    [5.83e-15 , 9.34e-02],
    [2.32e-16 ,- 1.64e-18],
    [1.59e-30 ,- 7.92e-13],
    [-9.39e-23,  3.51e-19],
    [5.83e-15 , 9.34e-02],
    [2.32e-16 ,- 1.64e-18],
    [1.59e-30 ,- 7.92e-13],
    [-9.39e-23,  3.51e-19],
    [5.83e-15 , 9.34e-02],
    [2.32e-16 ,- 1.64e-18],
    [1.59e-30 ,- 7.92e-13]]
)

# hull = ConvexHull((np.unique(a,axis=0)))
hull = ConvexHull(a)

```

### Error message:
ssh://ncarrara@smithers.gw:22/home/sequel/ncarrara/anaconda3/bin/python -u /home/sequel/ncarrara/phd_code/ncarrara/sandbox/raise_free_npe.py
free(): invalid pointer
bash: line 1: 47562 Aborted                 (core dumped) env "PYCHARM_HOSTED"="1" "PYTHONUNBUFFERED"="1" "PYTHONIOENCODING"="UTF-8" "PYCHARM_MATPLOTLIB_PORT"="34287" "JETBRAINS_REMOTE_RUN"="1" "PYTHONPATH"="/home/sequel/ncarrara/phd_code:/home/sequel/ncarrara/gym_pydial:/home/ncarrara/courses:/home/ncarrara/website:/home/ncarrara/phd:/home/ncarrara/5c62ca1e5fa4e54cd32d4d24:/home/ncarrara/thesis:/home/sequel/ncarrara/.pycharm_helpers/pycharm_matplotlib_backend" '/home/sequel/ncarrara/anaconda3/bin/python' '-u' '/home/sequel/ncarrara/phd_code/ncarrara/sandbox/raise_free_npe.py'

Process finished with exit code 134 (interrupted by signal 6: SIGABRT)

BUT ALSO

ssh://ncarrara@smithers.gw:22/home/sequel/ncarrara/anaconda3/bin/python -u /home/sequel/ncarrara/phd_code/ncarrara/sandbox/raise_free_npe.py
bash: line 1: 26567 Segmentation fault      (core dumped) env "PYCHARM_HOSTED"="1" "PYTHONUNBUFFERED"="1" "PYTHONIOENCODING"="UTF-8" "PYCHARM_MATPLOTLIB_PORT"="34287" "JETBRAINS_REMOTE_RUN"="1" "PYTHONPATH"="/home/sequel/ncarrara/phd_code:/home/sequel/ncarrara/gym_pydial:/home/ncarrara/courses:/home/ncarrara/website:/home/ncarrara/phd:/home/ncarrara/5c62ca1e5fa4e54cd32d4d24:/home/ncarrara/thesis:/home/sequel/ncarrara/.pycharm_helpers/pycharm_matplotlib_backend" '/home/sequel/ncarrara/anaconda3/bin/python' '-u' '/home/sequel/ncarrara/phd_code/ncarrara/sandbox/raise_free_npe.py'

Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)

```
free(): invalid pointer

BUT ALSO

Segmentation fault      (core dumped) 

```

### Scipy/Numpy/Python version information:
```
1.1.0 1.15.4 sys.version_info(major=3, minor=7, micro=1, releaselevel='final', serial=0)
```

**Comment by tylerjereddy** [comment]
I can reproduce on master, but there are probably some issues with precision here and / or the question of "general position" for the generators.

If I tell Qhull to randomly (& slightly) "joggle" the generators it will alleviate the crash with or without filtering for unique values beforehand: `hull = ConvexHull(a, qhull_options="QJ")`

That said, probably a good idea to make sure those generators are correct given the high level of degeneracy. Only 4 of those generators are unique, and even after the filtering the floating point values are indeed quite small. Qhull precision docs: http://www.qhull.org/html/qh-impre.htm

**Comment by ncarrara** [comment]
Hi, thanks for your reply. That is indeed my dirty quick fix. Optimally, I'd expect to get a python exception, like the one raised when there are colinearity.

**Comment by pv** [comment]
Appears to be fixed by #10413
